server/controllers/request.js

```diff
@@ -99,3 +101,65 @@ exports.updateRequest = asyncHandler(async (req, res, next) => {
     }
   });
 });
+
+exports.payPetSitter = asyncHandler(async (req, res, next) => {
+  let { 
+    userId, 
+    customerId, 
+    sitterId,
+    email, 
+    name, 
+    serviceId, 
+    hourlyRate, 
+    hours,
+    successUrl,
+    cancelUrl
+  } = req.body;
+
+  const orderId = req.params.id;
+
+  try {  
+    if (customerId === "") {
+      const customer = await stripe.customers.create({email, name});
+      await Profile.updateOne({_id: userId}, {customerId: customer.id});
+      customerId = customer.id;
+    }
+  
+    if (serviceId === "") {
+      const service = await stripe.products.create({
+        name: "Pet sitting",
+        unit_label: "Hour(s)",
+      });
+      await Profile.updateOne({_id: sitterId}, {serviceId: service.id});
+      serviceId = service.id;
+    };
```

Comment:
This is a potential flaw in our logic, if someone malicious intentionally hits without serviceId then they can bloat our Stripe products with unnecessary products.

A better option is just making sure there is a product available that we've created and then returning an error if it cannot be retrieved.

Reply:
Hi Ethan. Should I create a new Stripe product when user becomes a sitter? So, doing it that way then when a customer wants to pay, the sitter information stored in the DB provides the product ID.  Please let me know if its right and I do this change. Thanks.